overlord/servicestate/servicemgr.go

```diff
@@ -62,6 +62,13 @@ func Manager(st *state.State, runner *state.TaskRunner) *ServiceManager {
 	// TODO: undo handler
 	runner.AddHandler("quota-control", m.doQuotaControl, nil)
 
+	// A separate, purpose specific task to handle support for adding
+	// a snap to a quota group upon install. We can't directly refer to
+	// the servicestate package, so this task takes care of calling quotaUpdate
+	// with the correct setup. This task also supports proper handling of
+	// failure during install and correctly removes the snap again.
+	runner.AddHandler("quota-on-install", m.doAddSnapToQuota, m.undoAddSnapToQuota)
```

Comment:
is strange to be specifc about install here, as the task could be reused differently late.  Maybe if we want to make it clear that this is about quotas first:

"quota-add-snap",  doQuotaAddSnap", undoQuotaAddSnap

Reply:
Ah ok, that makes sense, I had a lot trouble coming up with namings for the task and it's parameters, so I really appreciate the input here.